modules/lang-painless/src/main/java/org/opensearch/painless/PainlessScriptEngine.java

```diff
@@ -195,11 +195,12 @@ private <T> Type generateStatefulFactory(Loader loader, ScriptContext<T> context
             }
         }
 
-        for (int count = 0; count < newFactory.getParameterTypes().length; ++count) {
+        final Class<?>[] parameterTypes = newFactory.getParameterTypes();
+        for (int count = 0; count < newFactory.getParameterCount(); ++count) {
```

Comment:
Minor, probably `parameterTypes.length` is more appropriate here, instead of `newFactory.getParameterCount()`

Reply:
done